doc/specs/#11000 - Marks/Shell-Integration-Marks.md

```diff
@@ -0,0 +1,431 @@
+have us add the command executed. That's not perfect (we wouldn't be able to get
+error information), but it does work well enough. 
+
+```cmd
+TODO! Mike add a sample PROMPT that uses these marks approproately. 
+```
+
+### Gutter icons
+
+VsCode implements a set of gutter icons to the left of the buffer lines, to
+provide a UI element for exposing some quick actions to perform, powered by
+shell integartion.
+
+Gutter icons don't need to implement app-level actions at all. They _should_ be
+part of the control. At least, part of the UWP `TermControl`. These are some
+basic "actions" we could add to that menu. Since these are all attached to a
+mark anyways, we already know what mark the user interacted with, and where the
+start/end already is.
+* Copy command
+* Copy output
+* re-run command
+
+If we do decide to allow comments in marks (ala "bookmarks"), then we could use
+the gutter flyout to display the comment, and have the tooltip display that
+comment.
+
+### Showing the gutter
+
+> **warning**: TODO!, for discussion: how do we really want to do this? Just
+> stick it in the margin/padding? Or have it be a separate space in the "buffer"
+> If it's in the buffer itself, we can render it with the renderer, which by all
+> accounts, we probably should.
+
+## Settings proposals
+
+The below are the proposed additions to the settings for supporting marks and
+interacting with them. Some of these have already been added as experimental
+settings - these would be promoted to no longer be experimental. 
+
+Many of the sub-points on these settings are definitely "Future Consideration"
+level settings. For example, the `scrollToMark` `"highlight"` property. That one
+is certianly not something we need to ship with.  
+
+### Actions
+
+In addition to driving marks via the output, we will also want to support adding
+marks manually. These can be thought of like "bookmarks" - a user indicated
+region that means something to the user.
+
+* [ ] `addMark`: add a mark to the buffer. If there's a selection, use place the
+  mark covering at the selection. Otherwise, place the mark on the cursor row.
+  - [x] `color`: a color for the scrollbar mark. (in [#12948])
+  - [ ] `category`: one of `{"prompt", "error", "warning", "success", "info"}`
+  - [ ] `dismissSelection`: `bool`, default `true`. If true, dismiss the
+    selection (if there is one) after marking it. If false, leave the text
+    selected.
+    - This is considered for something like pattern matching, where a user might
+      want to match on "error: ", mark it with an error mark, and then also
+      colorize it. 
+* [ ] `scrollToMark`
+  - [x] `direction`: `["first", "previous", "next", "last"]`  (in [#12948])
+  - [ ] `select`: `bool`, default false. Select the text of the mark when it's scrolled to
+    - This might be better served by `selectCommand` / `selectOutput`, below. 
+  - [ ] [#13455] - `highlight`: `bool`, default false. Display a temporary

```

Comment:
> all animations should be enabled by default

That's an undoubtably spicy take 🤣 I'm okay with it being true, since it's easily disable-able, but we can discuss more when (if) we get to #13455